### PR TITLE
Mint module GenesisState, proto, and v3 no-op migration

### PR DESCRIPTION
**Pull request by relyt29**
## Purpose of Changes and their Description

As we're going through the v0.7.0 upgrade checklist I observed that PR #690 was missing `GenesisState` updates for the new keeper field. And since the errors are updated, AND since there's a new keeper field it probably makes sense to bump the `ConsensusVersion` of the mint module. I also added a migration for the mint module which is a no-op, just a print line that tells you that the mint module was upgraded!

## Link(s) to Ticket(s) or Issue(s) resolved by this PR

PROTO-2787

## Are these changes tested and documented?

These tests will have to be tested by hand as part of the v0.7.0 checklist upgrade flow.

### CODE REVIEW

**Review by amimart — DISMISSED** [review]
Good catch 😉

File: x/mint/types/genesis.go [hunk 1]
```diff
@@ -1,19 +1,23 @@
 package types
 
-import "cosmossdk.io/math"
+import (
```
Comment (xmariachi) [inline x/mint/types/genesis.go]: single-line import instead?

File: x/mint/types/genesis.go [hunk 2]
```diff
@@ -42,5 +47,14 @@ func ValidateGenesis(data GenesisState) error {
 		return ErrInvalidEcosystemTokensMinted
 	}
 
+	if data.MonthsUnlocked.IsNegative() {
+		return ErrInvalidMonthsUnlocked
+	}
+
+	thirtySix := math.NewInt(36)
+	if data.MonthsUnlocked.GT(thirtySix) {
```
Comment (xmariachi) [inline x/mint/types/genesis.go:53]: MonthsUnlocked is defined as an int. Validation could also check < 0 values

Comment (relyt29) [reply x/mint/types/genesis.go:53]: literally two lines above it checks `IsNegative`

File: x/mint/types/genesis.go (hunk 1 above)
Comment (relyt29) [reply x/mint/types/genesis.go]: uhhh yeah

File: x/mint/module/module.go [hunk 3]
```diff
@@ -21,7 +21,7 @@ import (
 )
 
 // ConsensusVersion defines the current x/mint module consensus version.
-const ConsensusVersion = 2
+const ConsensusVersion = 3
```
Comment (kpeluso) [inline x/mint/module/module.go]: module mismatch. is it 3 or 4?

**Review by kpeluso — CHANGES_REQUESTED** [review]

**Review by kpeluso — APPROVED** [review]

**Review by amimart — APPROVED** [review]

**Pull request merged**